src/directory_sync/types/directory_group.rs

```diff
@@ -0,0 +1,119 @@
+use std::fmt::Display;
+
+use serde::{Deserialize, Serialize};
+
+use crate::{RawAttributes, Timestamps};
+
+//use crate::Timestamps;
+
+/// The ID of a [`Directory Group`].
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Serialize, Deserialize)]
+pub struct DirectoryGroupId(String);
+
+impl Display for DirectoryGroupId {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self.0)
+    }
+}
+
+impl From<String> for DirectoryGroupId {
+    fn from(value: String) -> Self {
+        Self(value)
+    }
+}
+
+impl From<&str> for DirectoryGroupId {
+    fn from(value: &str) -> Self {
+        Self(value.to_string())
+    }
+}
+
+/// The ID of a [`Directory`].
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Serialize, Deserialize)]
+pub struct DirectoryId(String);
+
+impl Display for DirectoryId {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self.0)
+    }
+}
+
+impl From<String> for DirectoryId {
+    fn from(value: String) -> Self {
+        Self(value)
+    }
+}
+
+impl From<&str> for DirectoryId {
+    fn from(value: &str) -> Self {
+        Self(value.to_string())
+    }
+}
+
+/// [WorkOS Docs: Directory](https://workos.com/docs/reference/directory-sync/directory)
+#[derive(Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub struct DirectoryGroup {
+    /// Unique identifier for the Directory Group.
+    pub id: DirectoryGroupId,
+
+    /// Unique identifier for the group, assigned by the Directory Provider.
+    /// Different Directory Providers use different ID formats.
+    pub idp_id: String,
+
+    /// The identifier of the [`Directory`] the Directory Group belongs to.
+    pub directory_id: DirectoryId,
+
+    /// The name of the Directory Group.
+    pub name: String,
+
+    /// The timestamps for the Directory Group.
+    #[serde(flatten)]
+    pub timestamps: Timestamps,
+
+    /// The raw attributes received from the Identity Provider.
+    pub raw_attributes: RawAttributes,
+}
+
+#[cfg(test)]
+mod test {
+    use super::{DirectoryGroup, DirectoryGroupId, DirectoryId};
+    use crate::{RawAttributes, Timestamps};
+    use chrono::DateTime;
+    use serde_json::{json, Value};
+    use std::collections::HashMap;
+    #[test]
```

Comment:
nit: Could we add a newline here between the imports and the test?

```suggestion

    #[test]
```